homeassistant/components/matter/config_flow.py

```diff
@@ -1,53 +1,208 @@
 """Config flow for Matter integration."""
 from __future__ import annotations
 
+import asyncio
 from typing import Any
 
 from matter_server.client.client import Client
 from matter_server.client.exceptions import CannotConnect, InvalidServerVersion
 import voluptuous as vol
 
 from homeassistant import config_entries
+from homeassistant.components.hassio import (
+    AddonError,
+    AddonInfo,
+    AddonManager,
+    AddonState,
+    HassioServiceInfo,
+    is_hassio,
+)
 from homeassistant.const import CONF_URL
 from homeassistant.core import HomeAssistant
-from homeassistant.data_entry_flow import FlowResult
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
 from homeassistant.helpers import aiohttp_client
 
-from .const import DOMAIN, LOGGER
-
-STEP_USER_DATA_SCHEMA = vol.Schema(
-    {
-        vol.Required(CONF_URL, default="http://172.30.32.1:5580/chip_ws"): str,
-    }
+from .addon import get_addon_manager
+from .const import (
+    ADDON_SLUG,
+    CONF_INTEGRATION_CREATED_ADDON,
+    CONF_USE_ADDON,
+    DOMAIN,
+    LOGGER,
 )
 
+ADDON_SETUP_TIMEOUT = 5
+ADDON_SETUP_TIMEOUT_ROUNDS = 40
+DEFAULT_URL = "ws://172.30.32.1:5580/chip_ws"
```

Comment:
can't we use the dns name here instead ?
when using supervisor each add-on has its own dns name based on slug

Reply:
The default url here will only be used in the manual step, ie not when using the official add-on. I'm not sure what's reasonable. Maybe just `localhost`?

Reply:
```suggestion
DEFAULT_URL = "ws://localhost:5580/chip_ws"
```